### PR TITLE
[aarch64] INTTERUPT_ defines are used in a switch which needs to resp…

### PR DESCRIPTION
**Pull request by kangaroo**
…ect c++11 narrowing

ARM is going to be a bit trickier, since it has 16 and 32-bit encodings -- tag @benpye you're it.

### CODE REVIEW

File: src/vm/gccover.cpp [hunk 1]
```diff
@@ -1134,7 +1134,16 @@ bool IsGcCoverageInterrupt(LPVOID ip)
...
     // Now it's safe to dereference the IP to check the instruction
+    // The instruction is of variable length depending on the architecture
+    // Check gccover.h and ensure that you switch on the proper length
+    // to avoid narrowing issues
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
     UINT8 instructionCode = *reinterpret_cast<UINT8 *>(ip);
+#elif defined(_TARGET_ARM64_)
+    UINT32 instructionCode = *reinterpret_cast<UINT32 *>(ip);
+#else
+#error Implement me for your architecture
```
Comment (jkotas) [inline src/vm/gccover.cpp]: The convention is to use `PORTABILITY_ASSERT` for these.

**Comment by jkotas** [comment]
> ARM is going to be a bit trickier, since it has 16 and 32-bit encodings -- tag @benpye you're it.

There is `GetARMInstructionLength`. It is used in number of other places in this file. The same logic just needs to be replicated here as well.

**Comment by danmosemsft** [comment]
@kangaroo checking in on this, is this PR still in progress?

**Comment by jkotas** [comment]
The issue has been fixed already.

**Pull request closed without merging**